doc/cookbook/resilience.md

```diff
@@ -0,0 +1,364 @@
+      policyRef: time-based-policy
+```
+
+In addition to failures, the circuit breaker can also short-circuit requests on slow requests. Below configuration regards requests which cost more than 30 seconds as slow requests and short-circuits requests if 60% of recent requests are slow.
+
+```yaml
+    policies:
+    - name: count-based-policy
+      slowCallRateThreshold: 60
+      slowCallDurationThreshold: 30s
+```
+
+For a policy, if the first request fails, the failure rate could be 100% because there's only one request. This is not the desired behavior in most cases, we can avoid it by specifying `minimumNumberOfCalls`.
+
+```yaml
+    policies:
+    - name: count-based-policy
+      minimumNumberOfCalls: 10
+```
+
+We can also configure the wait duration in open state, and the max wait duration in half-open state:
+
+```yaml
+    policies:
+    - name: count-based-policy
+      waitDurationInOpenState: 2m
+      maxWaitDurationInHalfOpenState: 1m
+```
+
+In half-open state, we can limit the number of permitted requests:
+
+```yaml
+    policies:
+    - name: count-based-policy
+      permittedNumberOfCallsInHalfOpenState: 10
+```
+
+For the full YAML, see [here](#circuitbreaker-1)
+
+### RateLimiter
+
+> NOTE: When there are multiple instances of Easegress, the configuration will be applied for every instance equally. For example, TPS of RateLimiter is configured with 100 in 3-instances cluster, so the total TPS will be 300.
+
+The below configuration limits the request rate for requests to `/admin` and requests that match regular expression `^/pets/\d+$`.
+
+```yaml
+name: pipeline-reverse-proxy
+kind: HTTPPipeline
+flow:
+  - filter: rate-limiter
+  - filter: proxy
+filters:
+  - name: rate-limiter
+    kind: RateLimiter
+    policies:
+    - name: policy-example
+      timeoutDuration: 100ms
+      limitRefreshPeriod: 10ms
+      limitForPeriod: 50
+    defaultPolicyRef: policy-example
+    urls:
+    - methods: [GET, POST, PUT, DELETE]
+      url:
+        exact: /admin
+        regex: ^/pets/\d+$
+      policyRef: policy-example
+  - name: proxy
+    kind: Proxy
+```
+
+For the full YAML, see [here](#ratelimiter-1)
+
+### Retryer
+
+If we want to retry on HTTP status code 500, 503, and 504, we can create a `Retryer` with below configuration, it makes at most 3 attempts on failure.
+
+```yaml
+name: pipeline-reverse-proxy
+kind: HTTPPipeline
+flow:
+  - filter: retryer
+  - filter: proxy
+filters:
+  - name: retryer
+    kind: Retryer
+    policies:
+    - name: policy-example
+      maxAttempts: 3
+      waitDuration: 500ms
+      failureStatusCodes: [500, 503, 504]
+    defaultPolicyRef: policy-example
+    urls:
+    - methods: [GET, POST, PUT, DELETE]
+      url:
+        prefix: /books/
+      policyRef: policy-example
+  - name: proxy
+    kind: Proxy
+```
+
+By default, the wait duration between two attempts is `waitDuration`, but this can be changed by specify `backOffPolicy` and `randomizationFactor`.
```

Comment:
```suggestion
By default, the wait duration between two attempts is `waitDuration`, but this can be changed by specifying `backOffPolicy` and `randomizationFactor`.
```